cpp/cmake/thirdparty/get_nvcomp.cmake

```diff
@@ -0,0 +1,39 @@
+#=============================================================================
+# Copyright (c) 2021, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#=============================================================================
+
+function(find_and_configure_nvcomp VERSION)
+
+    # Find or install nvcomp
+    rapids_cpm_find(nvcomp ${VERSION}
+        GLOBAL_TARGETS     nvcomp::nvcomp
+        CPM_ARGS
+            GITHUB_REPOSITORY  NVIDIA/nvcomp
+            GIT_TAG            3a12516afdeab4ace01298031757f84b8dda81b7
+            # GIT_SHALLOW        TRUE
+            OPTIONS            "BUILD_TESTS OFF"
```

Comment:
We should add `BUILD_STATIC ON` to the nvcomp options. This will make it build as a static library and become embedded into cudf so we don't need to distribute another shared library along side cudf

Reply:
Will the nvcomp APIs be exported to users of libcudf.so?  We have Java bindings to nvcomp in the cudf Java bindings (yeah, it's odd, long story).  It would be nice to be able to leverage the lone instance of nvcomp rather than shipping a separate copy of nvcomp in the cudf jar.

Reply:
If built statically we wouldn't be able to provide a stable nvcomp API since any function not used by libcudf.so would be removed by the linker. 

If we need to package nvcomp as part of our dependencies we need to make sure of the following:

- we are using the same nvcomp between the java bindings and cudf
- we have a conda nvcomp package

Reply:
Yeah if it's complicated we'll just ship our own statically-linked nvcomp with the Java bindings.